src/test/Fuzzers.sol

```diff
@@ -10,35 +10,95 @@ import {BalanceDelta} from "../types/BalanceDelta.sol";
 import {TickMath} from "../libraries/TickMath.sol";
 import {Pool} from "../libraries/Pool.sol";
 import {PoolModifyLiquidityTest} from "./PoolModifyLiquidityTest.sol";
+import {LiquidityAmounts} from "../../test/utils/LiquidityAmounts.sol";
+import {SafeCast} from "../../src/libraries/SafeCast.sol";
 
 contract Fuzzers is StdUtils {
+    using SafeCast for uint256;
+
     Vm internal constant _vm = Vm(address(uint160(uint256(keccak256("hevm cheat code")))));
 
-    function boundLiquidityDelta(PoolKey memory key, int256 liquidityDelta) internal pure returns (int256) {
-        return bound(
-            liquidityDelta, 0.0000001e18, int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)) / 2)
+    function boundLiquidityDelta(PoolKey memory key, int256 liquidityDeltaUnbounded, int256 liquidityMaxByAmount)
+        internal
+        pure
+        returns (int256)
+    {
+        int256 liquidityMaxPerTick = int256(uint256(Pool.tickSpacingToMaxLiquidityPerTick(key.tickSpacing)));
+
+        // Finally bound the seeded liquidity by either the max per tick, or by the amount allowed in the position range.
+        int256 liquidityMax = liquidityMaxByAmount > liquidityMaxPerTick ? liquidityMaxPerTick : liquidityMaxByAmount;
+        return bound(liquidityDeltaUnbounded, 0.0000001e18, liquidityMax);
```

Comment:
whats the reason for bounding by `0.0000001e18`?

Reply:
We just don't want 0 - because thats collecting fees. So we want the smallest amount of liquidity we can add without it being 0

Reply:
and why not just `1`?

Reply:
1 would work, this code is just taking the most extremes for minimum and more extremes for maximum

Reply:
I'd suggest using 1, the tests still pass and we have a broader range of values